cypress/support/sessions.js

```diff
@@ -50,16 +50,40 @@ Cypress.Commands.add('pushSteps', ({ connection, steps, version, awareness = ''
 		.then(response => response.data)
 })
 
+Cypress.Commands.add('failToPushSteps', ({ connection, steps, version, awareness = '' }) => {
+	return connection.push({ steps, version, awareness })
+		.then((response) => {
+			throw new Error('Expected request to fail - but it succeeded!')
+		})
+		.catch((err) => err.response)
```

Comment:
I think the `throw` in `then` will be caught in the catch and cause a `undefined` to be returned as `err.response` is undefined for the exception thrown.
You probably want to handle both cases in one [`then(onFullfilled, onRejected)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#syntax) call:
```suggestion
		.then(
			() => throw new Error('Expected request to fail - but it succeeded!'),
			err => err.response
		)
```

Reply:
Indeed, sounds logical. Changed now.